internal/provider/resource_vault_cluster.go

```diff
@@ -176,24 +189,54 @@ func resourceVaultClusterCreate(ctx context.Context, d *schema.ResourceData, met
 
 	publicEndpoint := d.Get("public_endpoint").(bool)
 
+	// ensure Plus tier performance replication params are legit
+	diagErr, primaryClusterModel := validatePerformanceReplicationChecksAndReturnPrimaryIfAny(ctx, client, d)
+	if diagErr != nil {
+		return diagErr
+	}
+
 	log.Printf("[INFO] Creating Vault cluster (%s)", clusterID)
 
-	vaultCuster := &vaultmodels.HashicorpCloudVault20201125InputCluster{
-		Config: &vaultmodels.HashicorpCloudVault20201125InputClusterConfig{
-			VaultConfig: &vaultmodels.HashicorpCloudVault20201125VaultConfig{
-				InitialVersion: vaultVersion,
+	var vaultCluster *vaultmodels.HashicorpCloudVault20201125InputCluster
+	if getPrimaryLinkIfAny(d) != "" {
+		// performance replication secondary cluster creation request.
```

Comment:
I might drop this or reword.
```suggestion
		If there is a primary link, create a performance secondary cluster.
```